src/libstore/local-store.cc

```diff
@@ -344,7 +360,11 @@ LocalStore::LocalStore(const Params & params)
     else openDB(*state, false);
 
     if (experimentalFeatureSettings.isEnabled(Xp::CaDerivations)) {
-        migrateCASchema(state->db, dbDir + "/ca-schema", globalLock);
+        if (!readOnly) {
+            migrateCASchema(state->db, dbDir + "/ca-schema", globalLock);
+        } else {
+            throw Error("need to migrate to CA schema, but this cannot be done in read-only mode");
```

Comment:
```suggestion
            throw Error("need to migrate to content-addressed schema, but this cannot be done in read-only mode");
```

Do not use abbreviations, this would be recursive jargon and very hard to parse for non-experts.